[v11] Backport of dependabot CVE updates

jentfoo opened the pull request:
Cherry-picked security updates that have been applied in master but failed to backport to v11

jentfoo commented on api/go.mod:
```diff
@@ -19,10 +19,11 @@ require (
-	golang.org/x/exp v0.0.0-20230105202349-8879d0199aa3
-	golang.org/x/net v0.0.0-20220927171203-f486391704dc
-	google.golang.org/grpc v1.49.0
+	golang.org/x/crypto v0.2.0 // replaced
```
As part of the cherry-picking it looks like the replacement was removed.  My plan is to clean these details up in an additional commit, so I will remove the comment as part of that.  Thank you!

codingllama commented on go.mod:
```diff
@@ -4,8 +4,8 @@ go 1.19
 	cloud.google.com/go/container v1.7.0
-	cloud.google.com/go/firestore v1.6.1
-	cloud.google.com/go/iam v0.6.0
+	cloud.google.com/go/firestore v1.9.0
```
Maybe move Firestore to a separate PR? It's a bigger and more delicate change than the others.

jentfoo replied:
I believe this change can be reverted.  It came in as part of the dependabot-batcher PR, but is not related to a CVE

zmb3 replied:
+1

jentfoo replied:
Reverted, there are still a good amount of other updates included that are not CVE related.  I am seeing if the build passes, but feel free to call out any others that seem potentially risky.  Thank you!

jentfoo replied:
After the unit test failures I agree.  Let's land the already minimal #23577 and use that as a reference for what needs updated in here and v12

codingllama replied:
Your call. Not every change on release branches needs to be a backport, so feel free to start this from a clean slate if that's better.

jentfoo committed:
[v10] Backport of dependabot CVE updates (#23577)

* Bump golang.org/x/net in /assets/backport (#22354)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.0.0-20210226172049-e18ecbb05110 to 0.7.0.
- [Release notes](https://github.com/golang/net/releases)
- [Commits](https://github.com/golang/net/commits/v0.7.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump golang.org/x/net in /web/.cloudbuild/scripts (#22353)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.0.0-20211216030914-fe4d6282115f to 0.7.0.
- [Release notes](https://github.com/golang/net/releases)
- [Commits](https://github.com/golang/net/commits/v0.7.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump golang.org/x/crypto in /assets/backport (#22359)

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.0.0-20210817164053-32db794688a5 to 0.1.0.
- [Release notes](https://github.com/golang/crypto/releases)
- [Commits](https://github.com/golang/crypto/commits/v0.1.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump golang.org/x/crypto in /web/.cloudbuild/scripts (#22360)

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.0.0-20211215153901-e495a2d5b3d3 to 0.1.0.
- [Release notes](https://github.com/golang/crypto/releases)
- [Commits](https://github.com/golang/crypto/commits/v0.1.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump golang.org/x/net from 0.6.0 to 0.7.0 in /api (#22010)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.6.0 to 0.7.0.
- [Release notes](https://github.com/golang/net/releases)
- [Commits](https://github.com/golang/net/compare/v0.6.0...v0.7.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump golang.org/x/net from 0.6.0 to 0.7.0 (#22012)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.6.0 to 0.7.0.
- [Release notes](https://github.com/golang/net/releases)
- [Commits](https://github.com/golang/net/compare/v0.6.0...v0.7.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Downgrade cyrpto to avoid bcrypt 72char limit change

* Go mod update revert

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jentfoo commented:
@codingllama, @rosstimothy, @zmb3, this PR has now been reduced to a more minimal set of updates.  Can you please review this?  Thank you!

jentfoo commented on assets/backport/go.mod:
```diff
@@ -17,10 +17,10 @@ require (
-	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110 // indirect
-	golang.org/x/sys v0.0.0-20210615035016-665e8c7367d1 // indirect
-	golang.org/x/term v0.0.0-20201126162022-7de9c90e9dd1 // indirect
+	golang.org/x/crypto v0.1.0 // indirect
+	golang.org/x/net v0.7.0 // indirect
+	golang.org/x/sys v0.5.0 // indirect
+	golang.org/x/term v0.5.0 // indirect
```
Typically I would, but with the goal of keeping these updates minimal I am not currently.  I suspect that these API's are stable enough that this should be the safest route, but if there is a preference for a greater change I can update the root dependencies depending on these.

codingllama commented on assets/backport/go.mod:
```diff
@@ -17,10 +17,10 @@ require (
-	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110 // indirect
-	golang.org/x/sys v0.0.0-20210615035016-665e8c7367d1 // indirect
-	golang.org/x/term v0.0.0-20201126162022-7de9c90e9dd1 // indirect
+	golang.org/x/crypto v0.1.0 // indirect
```
Bump to 0.2.0 while we're at it?

Should we do the same for web/.cloudbuild/scripts/go.mod as well?

jentfoo committed:
Update crypto to 0.2.0 universally
